pkg/services/ngalert/eval/cache.go

```diff
@@ -0,0 +1,102 @@
+package eval
```

Comment:
I wonder if this cache should be in datasources package. It is generic enough to be reusable